docs/content/3.cookbook/2.pass-custom-headers-and-cookies-to-requests.md

```diff
@@ -0,0 +1,266 @@
+---
+title: Pass Custom Headers and Cookies to Requests
+layout: default
+scope: sdk, middleware, integrations
+---
+
+# Pass Custom Headers and Cookies to Requests
+
+In this tutorial, we'll demonstrate how to set up an interceptor to pass headers when making requests using the axios client inside the Vue Storefront SDK. This is useful when you need to append custom headers or cookies to your API calls.
+
+After passing the headers to the SDK client, we'll also show you how to pass them further from the Server Middleware to external services (e.g. SAP OCC API).
+
+## Notes
+::info
+This guide uses the integrations boilerplate with the nuxt option. The same basic principles apply to other frameworks, but the implementation may vary.
+::
+
+All SDK modules export a client instance of axios. This is the axios instance that is used to make requests to the Server Middleware. You can access this instance by importing the `client` from the SDK.
+
+## Steps
+
+step 1.
+###  **Import Required Modules**
+
+```javascript
+import { initSDK, buildModule } from '@vue-storefront/sdk';
+import { client, boilerplateModule, BoilerplateModuleType } from '../../packages/sdk/src';
+import useHeaders from 'path-to-useHeaders-file'; // Replace with the correct path to the `useHeaders` composable.
```

Comment:
I don't understand, who does provide the `useHeaders`?

Reply:
It also feels like it's strictly vue specific, is this what we want?

Reply:
I see that the customer is responsible for creation of the composable, but my question regarding the vue-centric docs remain.

Reply:
I agree, if we can create a solution that  framework agnostic that would be ideal. But I think given the differences in how Next and Nuxt handle ssr/state, I'm not sure that's possible. I'm not familiar enough with Next to know. 

For context, the guide was made in response to an integrator's query that was using Nuxt. This started as general enablement and turned into a guide when several more folks asked the same question and all were using Nuxt. 

If it's not possible to provide a framework agnostic solution, we can at least add examples in both support frameworks.

Reply:
I'm ok for both frameworks! :) Just including nuxt only solution felt a bit off